includes/api/class-popups-analytics-controller.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Popups Analytics API endpoints.
+ *
+ * @package Newspack\API
+ */
+
+namespace Newspack\API;
+
+use \WP_REST_Controller;
+use \WP_Error;
+
+defined( 'ABSPATH' ) || exit;
+
+require_once NEWSPACK_ABSPATH . 'includes/popups-analytics/class-popups-analytics-utils.php';
+
+/**
+ * REST API endpoints for managing wizards.
+ */
+class Popups_Analytics_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = NEWSPACK_API_NAMESPACE;
+
+	/**
+	 * Endpoint resource.
+	 *
+	 * @var string
+	 */
+	protected $resource_name = 'popups-analytics';
+
+	/**
+	 * Register the routes.
+	 */
+	public function register_routes() {
+		// Register newspack/v1/wizards/some-wizard endpoint.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->resource_name . '/report',
+			[
+				[
+					'methods'             => 'GET',
+					'callback'            => [ $this, 'get_report' ],
+					'permission_callback' => [ $this, 'get_report_permissions_check' ],
```

Comment:
It's useful to enumerate and sanitize `args` for all endpoints, even if there's no real risk of abuse. At minimum it makes it a bit easier to troubleshoot problems.

Reply:
Done in 4d206950f3e0e908c8cfeabe9b06cb5b0918dc85